### PR TITLE
split into model & view

### PR DESCRIPTION
**Pull request by moloko**

### CODE REVIEW

File: js/assessmentResultsModel.js [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    var AssessmentResultsModel = ComponentModel.extend({
+
+        init: function() {
+            this.set('originalBody', this.get('body'));// save the original body text so we can restore it when the assessment is reset
+
+            this.listenTo(Adapt, {
+                'assessments:complete': this.onAssessmentComplete,
+                'assessments:reset': this.onAssessmentReset,
+                preRemove: function () {
```
Comment (oliverfoster) [inline js/assessmentResultsModel.js]: Odd change of style here with the quotes.

File: js/assessmentResultsModel.js [hunk 2]
```diff
@@ -0,0 +1,168 @@
+define([
+    'core/js/adapt',
+    'core/js/models/componentModel'
+], function(Adapt, ComponentModel) {
+    'use strict';
```
Comment (oliverfoster) [inline js/assessmentResultsModel.js]: Is the 'use strict' needed?

File: js/assessmentResultsModel.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+            this.toggleVisibility(true);
+        },
+
+        setFeedbackBand: function(state) {
+            var scoreProp = state.isPercentageBased ? 'scoreAsPercent' : 'score';
+            var bands = _.sortBy(this.get('_bands'), '_score');
+
+            for (var i = (bands.length - 1); i >= 0; i--) {
+                if (state[scoreProp] >= bands[i]._score) {
```
Comment (oliverfoster) [inline js/assessmentResultsModel.js]: continue early

File: js/assessmentResultsView.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+            this.model.checkIfAssessmentComplete();
+        },
+
+        postRender: function() {
+            this.setReadyStatus();
+        },
+
+        onInview: function(event, visible, visiblePartX, visiblePartY) {
+            if (visible) {
```
Comment (oliverfoster) [inline js/assessmentResultsView.js]: Return early would be cool.

File: js/assessmentResultsView.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            var assessmentId = this.model.get('_state').id;
+            Adapt.assessment.get(assessmentId).reset();
+        },
+
+        /**
+         * If there are classes specified for the feedback band, apply them to the containing article
+         * This allows for custom styling based on the band the user's score falls into
+         */
+        addClassesToArticle: function(model, value) {
+            if(!value._classes) return;
```
Comment (oliverfoster) [inline js/assessmentResultsView.js]: Space.

File: js/assessmentResultsModel.js (hunk 1 above)
Comment (moloko) [reply js/assessmentResultsModel.js]: https://github.com/adaptlearning/documentation/blob/master/01_cross_workstream/style_guide.md#object--array-creation

File: js/assessmentResultsModel.js (hunk 2 above)
Comment (moloko) [reply js/assessmentResultsModel.js]: It's not essential but has been included in the recent model/view updates so I've just been following that style

Comment (oliverfoster) [reply js/assessmentResultsModel.js]: I was included for the let/const stuff just never removed. It's entirely unnecessary as far as I can see.

File: js/assessmentResultsModel.js (hunk 1 above)
Comment (oliverfoster) [reply js/assessmentResultsModel.js]: Wow! I should have deleted that sentence rather than corrected it.

File: js/assessmentResultsModel.js (hunk 3 above)
Comment (moloko) [reply js/assessmentResultsModel.js]: Do you mean like this?
```js
for (var i = (bands.length - 1); i >= 0; i--) {
    if (state[scoreProp] < bands[i]._score) continue;

    this.set('_feedbackBand', bands[i]);
    break;
}
```
Just looks a bit clunky to my eyes - but if others prefer this style I'm happy to change

File: js/assessmentResultsModel.js (hunk 1 above)
Comment (oliverfoster) [reply js/assessmentResultsModel.js]: It is however a view render life-cycle event. So should be on the view, not the model.

File: js/assessmentResultsModel.js (hunk 3 above)
Comment (oliverfoster) [reply js/assessmentResultsModel.js]: I'd've gone one step further
```js
for (var i = (bands.length - 1); i >= 0; i--) {
    var isScoreInBandRange = (state[scoreProp] => bands[i]._score);
    if (!isScoreInBandRange) continue;

    this.set('_feedbackBand', bands[i]);
    break;
}
```

Comment (moloko) [reply js/assessmentResultsModel.js]: not sure about your use of the arrow function operator there - but I see what you mean ;-)

**Comment by moloko** [comment]
So... I bumped the version number to 3.0.0 but I'm starting to wonder if that's really necessary? Although this is obviously a big change to the code there are no breaking changes nor new features. Or could the model/view split be considered a 'feature'?

**Comment by oliverfoster** [comment]
I defer to stringfield, the master of the higgs fields.

**Comment by tomgreenfield** [comment]
Happy for a simple patch bump if it's just a behind-the-scenes refactor.

File: js/assessmentResultsModel.js [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+                'assessments:complete': this.onAssessmentComplete,
+                'assessments:reset': this.onAssessmentReset
+            });
+        },
+
+        /**
+         * Checks to see if the assessment was completed in a previous session or not
+         */
+        checkIfAssessmentComplete: function() {
+            if (!Adapt.assessment ||this.get('_assessmentId') === undefined) {
```
Comment (lc-thomasberger) [inline js/assessmentResultsModel.js]: very minor, but can you please add a space after `||`

**Pull request merged**